libraries/exoplayer_hls/src/main/java/androidx/media3/exoplayer/hls/playlist/DefaultHlsPlaylistTracker.java

```diff
@@ -695,6 +719,27 @@ private void loadPlaylistInternal(Uri playlistRequestUri) {
     }
 
     private void loadPlaylistImmediately(Uri playlistRequestUri) {
+      if (primaryMediaPlaylistUrl == null || playlistRequestUri.equals(primaryMediaPlaylistUrl)) {
+        SntpClient.InitializationCallback callback =
+            new SntpClient.InitializationCallback() {
+              @Override
+              public void onInitialized() {
+                loadAfterTimeSync(playlistRequestUri);
+              }
+
+              @Override
+              public void onInitializationFailed(IOException error) {
+                Log.w(TAG, "NTP time init failed, use default system time", error);
+                loadAfterTimeSync(playlistRequestUri);
+              }
+            };
+        SntpClient.initialize(mediaPlaylistLoader, callback);
+      } else {
+        loadAfterTimeSync(playlistRequestUri);
+      }
+    }
+
+    private void loadAfterTimeSync(Uri playlistRequestUri) {
       ParsingLoadable.Parser<HlsPlaylist> mediaPlaylistParser =
```

Comment:
The same refactor lift method to create `loadAfterTimeSync()`.  Here only the primary playlist load needs to possibly refresh `SntpClient` as the master playlist is only loaded once